plugins/api/src/register_physical_path.cpp

```diff
@@ -1346,16 +1350,19 @@ auto plugin_factory(const std::string& _instance_name,
 #endif // RODS_SERVER
 
     // clang-format off
-    irods::apidef_t def{REGISTER_PHYSICAL_PATH_APN,                 // API number
-                        RODS_API_VERSION,                           // API version
-                        NO_USER_AUTH,                               // Client auth
-                        NO_USER_AUTH,                               // Proxy auth
-                        "DataObjInp_PI", 0,                         // In PI / bs flag
-                        "BinBytesBuf_PI", 0,                        // Out PI / bs flag
-                        op,                                         // Operation
-                        "api_register_physical_path",               // Operation name
-                        nullptr,                                    // Null clear function
-                        (funcPtr) CALL_REGISTER_PHYSICAL_PATH};
+    irods::apidef_t def{
+        REGISTER_PHYSICAL_PATH_APN,     // API number
+        RODS_API_VERSION,               // API version
+        NO_USER_AUTH,                   // Client auth
+        NO_USER_AUTH,                   // Proxy auth
+        "DataObjInp_PI", 0,             // In PI / bs flag
+        "BinBytesBuf_PI", 0,            // Out PI / bs flag
+        op,                             // Operation
+        "api_register_physical_path",   // Operation name
+        clearDataObjInp,                // Clear input function // TODO Why was this set to NOP?
+        [](void* _p) { clearBBuf(static_cast<BytesBuf*>(_p)); }, // Clear output function
+        (funcPtr) CALL_REGISTER_PHYSICAL_PATH
+    };
```

Comment:
@alanking Notice the TODO.

Is it correct for this API plugin to not clear the incoming `DataObjInp`?

Reply:
This is fine as the clear function is only triggered when the rc-call is invoked.